base_station/gui/src/components/IKControls.vue

```diff
@@ -154,6 +174,29 @@ export default {
 
         presetPositionCallback: function(position) {
             this.$parent.publish('/arm_preset', { 'type': 'ArmPreset', 'preset': position });
+        },
+
+        addPreset: function() {
+            if (this.buttonInc == 1) {
+                this.button1 = this.name;
+                this.$parent.publish('/custom_preset', { 'type': 'CustomPreset', 'preset': this.button1 });
+            }
+            else if (this.buttonInc == 2) {
+                this.button2 = this.name;
+                this.$parent.publish('/custom_preset', { 'type': 'CustomPreset', 'preset': this.button2 });
+            }
+            else if (this.buttonInc == 3) {
+                this.button3 = this.name;
+                this.$parent.publish('/custom_preset', { 'type': 'CustomPreset', 'preset': this.button3 })
+            }
+            else if (this.buttonInc == 4) {
+                this.button4 = this.name;
+                this.$parent.publish('/custom_preset', { 'type': 'CustomPreset', 'preset': this.button4 });
+            }
+            else if (this.buttonInc == 5) {
+                this.button5 = this.name;
+                this.$parent.publish('/custom_preset', { 'type': 'CustomPreset', 'preset': this.button5 });
+            }
```

Comment:
We can just call publish once at the bottom and pass this.name instead of this.button<x> right? (To avoid code duplication)